changelogs/2.10.0.md

```diff
@@ -0,0 +1,933 @@
+  methods but allows to execute requests sequentially. ID is generated
+  automatically on the client side. If users write their own connector and
+  want to use streams, they must transmit stream_id over iproto protocol.
+  The main purpose of streams is transactions via iproto.
+  Each stream can start its own transaction, allowing multiplexing several
+  transactions over one connection. There are multiple ways to begin,
+  commit and rollback a transaction: using appropriate stream methods, using
+  `call` or `eval` methods or using `execute` method with SQL transaction
+  syntax. Users can mix these methods, for example, start a transaction using
+  `stream:begin()`, and commit a transaction using `stream:call('box.commit')`
+  or `stream:execute('COMMIT')`.
+
+  If any request fails during the transaction, it will not affect the other
+  requests in the transaction. If a disconnect occurs when there is some active
+  transaction in the stream, this transaction will be rolled back if it
+  does not have time to commit before this moment.
+* Added new `memtx_allocator` option to `box.cfg{}` that allows to select the
+  appropriate allocator for memtx tuples if necessary. Possible values are
+  `system` for malloc allocator and `small` for default small allocator.
+
+  Implemented system allocator, based on malloc: slab allocator that is used for
+  tuples allocation has a certain disadvantage---it tends to unresolvable
+  fragmentation on certain workloads (size migration). In this case, a user
+  should be able to choose another allocator. System allocator based on malloc
+  function, but restricted by the same quota as slab allocator. System
+  allocator does not alloc all memory at start. Instead, it allocates memory as
+  needed, checking that quota is not exceeded (gh-5419).
+* Added `box.stat.net.thread()` for reporting per thread net statistics
+  (gh-6293).
+* Added the new `STREAMS` metric to `box.stat.net` that contains statistics
+  for iproto streams. `STREAMS` contains same counters as `CONNECTIONS`
+  metric in `box.stat.net`: current, rps, and total (gh-6293).
+* Extended the network protocol (IPROTO) with a new request type (`IPROTO_ID`)
+  that is supposed to be used for exchanging sets of supported features between
+  server and client (gh-6253).
+* Added `required_protocol_version` and `required_protocol_features` to
+  `net.box` connection options. The new options allow to specify the IPROTO
+  protocol version and features that must be supported by the server for the
+  connection to pass (gh-6253).
+* **[Breaking change]** Added `msgpack.cfg.encode_error_as_ext` configuration
+  option to enable/disable encoding errors as `MP_ERROR` MsgPack extension.
+  The option is enabled by default (gh-6433).
+* **[Breaking change]** Removed `box.session.setting.error_marshaling_enabled`.
+  Error marshalling is now enabled automatically if the connector supports it
+  (gh-6428).
```

Comment:
Shouldn't breaking changes go first in the list?

Reply:
We outlined those changes in Compatibility section.